### PR TITLE
Enable oAuth login to UI

### PR DESCRIPTION
**Pull request by jimmidyson**
This currently authenticates against OpenShift specified in config file and
only allows access for an authenticated user.

Still have to retrieve user details from the oauth endpoint but that should be
pretty easy.

### CODE REVIEW

**Comment by jimmidyson** [comment]
This is working now, got user's name displaying in top right. Also includes the oauth token from OpenShift in the requests to iPaaS API server.

oauth

This uses module from https://github.com/manfredsteyer/angular-oauth2-oidc which will also work with Keycloak (module has OIDC discovery for easier configuration too).

**Comment by kahboom** [comment]
@jimmidyson - Is this good to go? Still have that question about the core feature module and shared feature component. Other than that and the unsightly OpenShift permissions screen (which we can't do much about), I've tested locally and am happy with this. 👍

**Comment by kahboom** [comment]
screenshot 2017-01-19 11 18 49

**Comment by jimmidyson** [comment]
From @liggitt re: permissions page:

> no, no plans to improve it beyond that currently
> not necessarily opposed, just nowhere near the top of the priority list

**Comment by jimmidyson** [comment]
retest this please

**Comment by jimmidyson** [comment]
retest this please

**Comment by jimmidyson** [comment]
retest this please

**Comment by jimmidyson** [comment]
retest this please

**Comment by jimmidyson** [comment]
retest this please

**Comment by jimmidyson** [comment]
retest this please

**Comment by jimmidyson** [comment]
retest this please

**Comment by jimmidyson** [comment]
retest this please

**Comment by jimmidyson** [comment]
retest this please

**Comment by jimmidyson** [comment]
Docker ftw! Jenkins runs tests in a Docker container now for each PR.

**Comment by jimmidyson** [comment]
retest this please

File: src/app/app.component.html [hunk 1]
```diff
@@ -36,7 +36,7 @@
         <!-- User Dropdown -->
         <li class="dropdown">
           <a class="dropdown-toggle nav-item-iconic" id="dropdownMenu2" data-toggle="dropdown" aria-haspopup="true" aria-expanded="true">
-            <span title="Username" class="fa pficon-user"></span>&nbsp;Clint Eastwood
+            <span title="Username" class="fa pficon-user"></span>&nbsp;{{(userService.user | async)?.fullName}}
```
Comment (kahboom) [inline src/app/app.component.html]: I'm gonna miss Clint Eastwood.

**Comment by kahboom** [comment]
Merging per conversation with Master Dyson.

**Pull request merged**